### PR TITLE
K3 PSCI Support

### PR DESCRIPTION
**Pull request by glneo**
Hello all,

This set enables basic PSCI support for the TI K3 platform.

The first patch is the Secure Proxy driver, Secure Proxy being a mailbox transport we use for communications with the system controller. The protocol for this communication is known as TI-SCI, introduced in the second patch. Then then next four build up the TI-SCI API across Device, Clock, Core, and Processor control. Lastly we add PSCI pwr_domain_on() and system_reset() functionality, additional commands will be added in subsequent patches.

Thanks,
Andrew

### CODE REVIEW

File: plat/ti/k3/common/drivers/sec_proxy/sec_proxy.c [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+ */
+#define SP_THREAD(_x) \
+	[_x] = { \
+		.id = _x, \
+		.data = SEC_PROXY_THREAD(SEC_PROXY_DATA_BASE, _x), \
+		.scfg = SEC_PROXY_THREAD(SEC_PROXY_SCFG_BASE, _x), \
+		.rt = SEC_PROXY_THREAD(SEC_PROXY_RT_BASE, _x), \
+	}
+
+struct k3_sec_proxy_mbox spm = {
```
Comment (antonio-nino-diaz-arm) [inline plat/ti/k3/common/drivers/sec_proxy/sec_proxy.c]: Can you make this static?

File: plat/ti/k3/common/drivers/ti_sci/ti_sci.c [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+ * struct ti_sci_info - Structure representing a TI SCI instance
+ * @desc:	SoC description for this instance
+ * @seq:	Seq id used for verification for tx and rx message
+ */
+struct ti_sci_info {
+	const struct ti_sci_desc desc;
+	uint8_t seq;
+};
+
+struct ti_sci_info info = {
```
Comment (antonio-nino-diaz-arm) [inline plat/ti/k3/common/drivers/ti_sci/ti_sci.c]: Can you make this static?

**Comment by glneo** [comment]
@antonio-nino-diaz-arm 
This statement was added to give credit to the Linux and U-Boot implementation authors, all of whom are TI employees, the versions they sent to those projects was GPL'd to meet those projects licencing guidelines, this implementation is BSD licenced and contains no GPL code.

File: plat/ti/k3/common/drivers/sec_proxy/sec_proxy.c (hunk 1 above)
Comment (glneo) [reply plat/ti/k3/common/drivers/sec_proxy/sec_proxy.c]: Will do.

File: plat/ti/k3/common/drivers/ti_sci/ti_sci.c (hunk 2 above)
Comment (glneo) [reply plat/ti/k3/common/drivers/ti_sci/ti_sci.c]: Will do.

**Comment by antonio-nino-diaz-arm** [comment]
Ok, no problem then.

IP Review +1

**Comment by jeenu-arm** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
The build/run tests have succeeded.

**Comment by antonio-nino-diaz-arm** [comment]
@glneo ping

**Comment by antonio-nino-diaz-arm** [comment]
This needs to be rebased because of the changes in libc.

**Comment by glneo** [comment]
Okay, rebased.

**Comment by glneo** [comment]
Ping?

**Comment by soby-mathew** [comment]
jenkins: test this please

**Comment by soby-mathew** [comment]
Looks good to me. Waiting for CI test results.

**Pull request merged**